### ISSUE TITLE
Installing on VMware

### ISSUE THREAD

**Issue by Draygrah** [ChrisTitusTech/ArchTitus#73]
Hi!

I'm trying to install this using the Arch Linux image on a VMware Workstation 16. The script runs through, and after reboot it only boots back in the installation media. If I remove the CD/DVD drive completely from the virtual machine, it tries to netboot and after a while, informs that PXE-M0F: Exiting Intel PXE ROM
Operating System not found.

Network boot is third option in the VM BIOS. Removable media is 1st, Hard disk 2nd.

How is this supposed to work on a virtual machine?

I would've loved to test this first on a vm, before wiping my main system. I previously installed Arch following LearnLinuxTV's guide on youtube, that was a fun day but I don't want to do it again. Now I have EndeavourOS installed, but it bothers me that it isn't Arch (well it is, but neofetch doesn't say Arch, and I want REAL Arch :D )

I'm hoping I can run this script to install Arch with ease, and maybe install the DistroTube's dtos script for the epic xmonad experience. Not sure how they work together, so I prefer testing it out on this vm first.

TL;DR: Please tell me how to install this on a VM.

**Comment by Fr3db3rt** [comment]
I had the same on ESXi ... change Boot Options from BIOS to UEFI did the trick

**Comment by Draygrah** [comment]
> I had the same on ESXi ... change Boot Options from BIOS to UEFI did the trick

Hmm.. The BIOS is "PhoenixBIOS", it doesn't have an option for UEFI. Can I get better BIOS for my VM somehow?

**Comment by Draygrah** [comment]
I now added a line:
firmware="efi"
to the .vmx file.

Apparently this should fix it. Now testing again.

**Comment by Draygrah** [comment]
![image](https://user-images.githubusercontent.com/60978680/139902060-1f26256e-f863-46c1-ae0e-3ca3a9a515e6.png)

**Comment by Draygrah** [comment]
Also tried to change the SCSI drive to SATA drive, same error.

**Comment by imoldovavan** [comment]
I am having the same problem with Oracle VM. No bootable medium found.

**Comment by imoldovavan** [comment]
Checking Enable EFI under System Motherboard tab fixed it.

**Comment by Draygrah** [comment]
> Checking Enable EFI under System Motherboard tab fixed it.

I don't have System Motherboard Tab in my VMPlayer :s

**Comment by Draygrah** [comment]
I'm running the script now again on a fresh install, and will try to:
1. shut down the vm
2. modify the .vmx manually to have the line firmware="efi"
3. start the vm and pray for the rng gods

**Comment by Draygrah** [comment]
Yeah, still doesn't work. I'm trying on VirtualBox now. If that doesn't work, I'll have to read about Docker and see if I can install it in a container. That's 100% new stuff to me, but should be interesting if I have to go there.

**Comment by imoldovavan** [comment]
VirtualBox works but for some reason I can't change the resolution. It snaps back when I hit the Apply button.

**Comment by Draygrah** [comment]
Yeah, Virtualbox boots up nicely. xrandr -s 1920x1080 sets correct resolution for 0,4 seconds, then snaps back to lower res. @imoldovavan 

@d4ve10 so you moved the lines 13-20 form 3-post-setup.sh to the 0-preinstall.sh. To where in there? End?

**Comment by d4ve10** [comment]
Moved the lines between line 108 and 109 in 0-preinstall.sh. You can see the changes [here](https://github.com/ChrisTitusTech/ArchTitus/pull/76/files). After the install, it should normally boot into grub.
Regarding the resolution problem:
Try to open the settings of the VM in VirtualBox, go to the Display section and change the Graphics Controller to something other than VMSVGA. VBoxVGA worked great for me.

**Comment by Draygrah** [comment]
Well, VirtualBox Crashed, and the install got corrupted (Boots only to the yellow EFI Shell). Scrapped it, and I'm now installing on VMWare yet again. 

This time I altered the scripts as @d4ve10 suggested. Fingers crossed...

I did read about Docker during my workday, bits and pieces. Shouldn't be hard to try that next.
